src/deprecate.jl

```diff
@@ -0,0 +1,4 @@
+Base.@deprecate(
+    BlockDiagonal{T, V}(blocks) where {T, V},
```

Comment:
why deprecate this rather than just computing S in this case?

Reply:
I was on the fence, but in the end it seemed simpler to have fewer constructors. If we decide to go with the suggestion on your other comment we should maybe bring this back?